src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ClassWriterImpl.java

```diff
@@ -197,7 +198,23 @@ public void addClassSignature(String modifiers, Content classInfoTree) {
         classInfoTree.add(hr);
         Content pre = new HtmlTree(TagName.PRE);
         addAnnotationInfo(typeElement, pre);
-        pre.add(modifiers);
+        String sep = null;
+        for (String modifiersPart : modifiers.split(" ")) {
+            if (sep != null) {
+                pre.add(sep);
+            }
+            if ("record".equals(modifiersPart) ||
+                "sealed".equals(modifiersPart) || "non-sealed".equals(modifiersPart)) {
+                pre.add(modifiersPart);
+                pre.add(new HtmlTree(TagName.SUP).add(HtmlTree.A("#preview", contents.previewMark)));
```

Comment:
This will likely clash with a public field named "preview". You should choose a name for the anchor that is not a Java identifier.